pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix

```diff
@@ -0,0 +1,135 @@
+{ mkDerivation, lib, fetchFromGitHub, fetchsvn, fetchpatch
+, pkgconfig, pythonPackages, cmake, ninja, gcc9, qtbase
+, qtimageformats, gtk3, libappindicator-gtk3, libnotify
+, xdg_utils, ffmpeg_4, openalSoft, lzma, lz4, zlib, minizip
+, openssl, libtgvoip, rlottie-tdesktop, range-v3
+}:
+
+with lib;
+
+mkDerivation rec {
+  pname = "kotatogram-desktop";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "kotatogram";
+    repo = "kotatogram-desktop";
+    rev = "k${version}";
+    sha256 = "156r229dpxnyaxxgp1dr4vz8jla5p7r0dydhgdgk2yx1p11z6m3w";
+    fetchSubmodules = true;
+  };
+
+  # Arch patches (svn export telegram-desktop/trunk)
+  archPatches = fetchsvn {
```

Comment:
You may want to put sources for patches in a separate `let ... in`.

Reply:
I can drop this with my patch for QtDBus-based notifications, so Kotatogram will not needed GTK. I just didn't except that you offer me to being a maintainer.

Reply:
I'm not saying you should drop this, all I'm saying is that unneeded arguments to `mkDerivation` are discouraged.

Reply:
I'm not saying you should drop this patch, all I'm saying is that extra unneeded arguments to mkDerivation are discouraged.

Reply:
Hmm, ok